src/Driver/Driver.hs

```diff
@@ -54,6 +55,15 @@ import Pretty.Program ()
 import Syntax.RST.Types qualified as RST
 
 
+addConstrForAnnot :: Maybe (RST.TypeScheme pol) -> TST.Term prdcns -> ConstraintSet -> ConstraintSet
+addConstrForAnnot pcdecl_annot tmInferred constraintSet =
+  case pcdecl_annot of 
+    Nothing -> constraintSet 
+    Just (RST.TypeScheme _ _ (RST.TyKindAnnot mk _)) -> addKindAnnotConstr (TST.getKind $ TST.getTypeTerm tmInferred) mk constraintSet
+    Just (RST.TypeScheme _ _ (RST.TyApp _ _ (RST.TyKindAnnot mk _) _)) -> addKindAnnotConstr (TST.getKind $ TST.getTypeTerm tmInferred) mk constraintSet
+    Just (RST.TypeScheme _ _ (RST.TyApp _ _ (RST.TyNominal _ _ pk _) _)) -> addKindAnnotConstr (TST.getKind $ TST.getTypeTerm tmInferred) (CBox $ returnKind pk) constraintSet
```

Comment:
Here I might wanna do some changes. Before this, I would only look at TyKindAnnot within the annotated type scheme, but now that TyNominal and TyApp are separate, they can't be annotated individually, which would give the syntax (Fun:CBV)(..) or (Fun)(((a:CBV),(b:CBV):CBV) neither of which work with the current parser and at least the second one also does not make sense.
So right now, the kind of nominal types is just always used when it's available, which would mean that a term def rec prd x : forall a.(a->a) := x would typecheck even without kind annotation